hail/python/hail/methods/qc.py

```diff
@@ -491,13 +507,335 @@ def n_discordant(counter):
+        return VEPConfig(
+            config['data_bucket'],
+            config['regions'],
+            config['image'],
+            config['data_mount'],
+            config['env'],
+            config['vep_json_typ'],
+            config['command'],
+            config['csq_header_command'],
+        )
+
+    def __init__(self,
+                 data_bucket: str,
+                 regions: List[str],
+                 image: str,
+                 data_mount: str,
+                 env: Dict[str, str],
+                 vep_json_typ: hl.expr.HailType,
+                 command: List[str],
+                 csq_header_command: List[str]):
+        self.data_bucket = data_bucket
+        self.regions = regions
+        self.image = image
+        self.data_mount = data_mount
+        self.env = env
+        self.vep_json_typ = vep_json_typ
+        self.command = command
+        self.csq_header_command = csq_header_command
+
+
+supported_vep_configs = {
+    ('GRCh37', 'gcp', 'us-central1', 'hail.is'): VEPConfig(
+        'hail-qob-vep-grch37-us-central1',
+        ['us-central1'],
+        'us-docker.pkg.dev/hail-vdc/hail/vep-grch37:cache-jigold',
+        '/vep_data/',
+        {'PERL5LIB': '/vep_data/loftee'},
+        VEPConfig.default_vep_json_typ,
+        ["python3", "/hail-vep/run_vep_grch37.py", "vep"],
+        ["python3", "/hail-vep/run_vep_grch37.py", "csq_header"],
+    ),
+}
+
+
+def supported_vep_config(backend: ServiceBackend, reference_genome: str, *, regions: Optional[List[str]] = None) -> VEPConfig:
+    cloud = backend.bc.cloud()
+    domain = configuration_of('global', 'domain', None, None)
+    possible_regions = backend.bc.supported_regions()
+    regions = configuration_of('batch', 'regions', regions, possible_regions)
+    if isinstance(regions, str):
+        regions = regions.split(',')
+
+    for region in regions:
+        config_params = (reference_genome, cloud, region, domain)
+        if config_params in supported_vep_configs:
+            return supported_vep_configs[config_params]
+
+    raise ValueError(f'could not find a supported vep configuration for reference genome {reference_genome}, '
+                     f'cloud {cloud}, regions {regions}, and domain {domain}')
+
+
+def _service_vep(backend: ServiceBackend,
+                 ht: Table,
+                 config: Optional[Dict[str, Any]],
+                 regions: List[str],
+                 block_size: int,
+                 csq: bool,
+                 tolerate_parse_error: bool):
+    reference_genome = ht['locus'].dtype.reference_genome.name
+    if config is not None:
+        vep_config = VEPConfig.from_dict(config)
+    else:
+        vep_config = supported_vep_config(backend, reference_genome, regions=regions)
+
+    token = secret_alnum_string(16)
+    vep_input_path = hl.TemporaryDirectory(prefix=f'vep/inputs/{token}.vcf')
+    vep_output_path = hl.TemporaryDirectory(prefix=f'vep/outputs/{token}')
```

Comment:
The token is unnecessary, temporary directory will get you a unique location using secret_alnum_string.